### PR TITLE
James/test startup - Added ChatTest spoofing of Chat hub correctly

### LINKED ISSUE #123: Write tests for Chat.cs
Working in Justine/persistedRepository to write basic function tests which incorporate the data models.
Expect to find some blockers along the way to correct which will hopefully smooth the transition with the most updated chat.cs working on SingalR (Heather-Kevin/chatroom)

### PR DESCRIPTION
**Pull request by JamesEarle**
Fixes #123 

 - You can now write tests for Chat.cs and spoof a SignalR hub correctly.
 - More tests will need to be written as we expand on Jabbr (and to get better code coverage on legacy functions) but this fixes the Travis build finally

### CODE REVIEW

**Comment by adamtuliper** [comment]
Need someone to test and :shipit:

File: src/JabbR-Core/Hubs/Chat.cs [hunk 1]
```diff
@@ -186,7 +187,7 @@ public void UpdateActivity()
                 UpdateActivity(user, room);
             }
 
-            CheckStatus();
+            CheckStatus(testing);
```
Comment (adamtuliper) [inline src/JabbR-Core/Hubs/Chat.cs]: Why are we allowing out of band version checks? Seems like this is something that we can easily forget to leave in and could be configured via config settings on the server side to allow out of band clients (only if env.IsDevelopment)

Comment (JamesEarle) [inline src/JabbR-Core/Hubs/Chat.cs]: The `CheckStatus` method checks versions, but the `Context.QueryString` object is null, and read only. I haven't found a way to spoof this yet, but the version checks are non-critical for testing so I just allowed calls to Send with an optional testing parameter that would skip over the test. Still working on this.

Comment (adamtuliper) [inline src/JabbR-Core/Hubs/Chat.cs]: ok, will merge, lets find a better mock way or other full integration test.

**Pull request merged**